zebrad/src/components/inbound.rs

```diff
@@ -344,6 +358,7 @@ impl Service<zn::Request> for Inbound {
                 // https://github.com/tower-rs/tower/blob/master/tower/src/util/call_all/common.rs#L112
                 use futures::stream::TryStreamExt;
                 hashes
+                    .clone()
```

Comment:
Minor: I'd suggest replacing `hashes.clone().into_iter()` with `hashes.iter().cloned()`, so that we avoid allocating space for a temporary container and maybe also avoid rehashing the items.

Reply:
We can get hundreds of thousands of these requests if another node is syncing from us. But most requests from `zebrad` only contain one block. (I'm not sure what `zcashd` does.)

So I don't know if this optimisation actually matters.

Fixed in e3914c957 in PR #3505